Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibsAPI.cs

```diff
@@ -0,0 +1,1006 @@
+        /// <param name="className">Document class name, as declared in the VBA project</param>
+        /// <param name="interfaceIIDs">An array of interface IIDs to check against</param>
+        /// <param name="matchedIndex">on return indicates the index into interfaceIIDs that matched</param>
+        /// <returns>bool indicating whether the class does inherit one of the specified interfaces</returns>
+        public static bool DoesClassImplementInterface(TypeLibWrapper projectTypeLib, string className, Guid[] interfaceIIDs, out int matchedIndex)
+        {
+            return DoesClassImplementInterface(projectTypeLib.TypeInfos.Get(className), interfaceIIDs, out matchedIndex);
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements one of several possible interfaces
+        /// </summary>
+        /// <param name="component">Safe-com wrapper representing the VBA component</param>
+        /// <param name="interfaceIIDs">An array of interface IIDs to check against</param>
+        /// <param name="matchedIndex">on return indicates the index into interfaceIIDs that matched</param>
+        /// <returns>bool indicating whether the class does inherit one of the specified interfaces</returns>
+        public static bool DoesClassImplementInterface(IVBComponent component, Guid[] interfaceIIDs, out int matchedIndex)
+        {
+            return DoesClassImplementInterface(component.ParentProject, component.Name, interfaceIIDs, out matchedIndex);
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements one of several possible interfaces
+        /// </summary>
+        /// <param name="classTypeInfo">Low-level ITypeInfo wrapper representing the VBA project</param>
+        /// <param name="interfaceIIDs">An array of interface IIDs to check against</param>
+        /// <param name="matchedIndex">on return indicates the index into interfaceIIDs that matched</param>
+        /// <returns>bool indicating whether the class does inherit one of the specified interfaces</returns>
+        public static bool DoesClassImplementInterface(TypeInfoWrapper classTypeInfo, Guid[] interfaceIIDs, out int matchedIndex)
+        {
+            return classTypeInfo.ImplementedInterfaces.DoesImplement(interfaceIIDs, out matchedIndex);
+        }
+
+        /// <summary>
+        /// Returns the class progID of a control on a UserForm
+        /// </summary>
+        /// <param name="ide">Safe-com wrapper representing the VBE</param>
+        /// <param name="projectName">VBA Project name, as declared in the VBE</param>
+        /// <param name="userFormName">UserForm class name, as declared in the VBA project</param>
+        /// <param name="controlName">Control name, as declared on the UserForm</param>
+        /// <returns>string class progID of the specified control on a UserForm, e.g. "MSForms.CommandButton"</returns>
```

Comment:
Earlier I commented the need to note what will happen if a control has mulitple interfaces, which one will "win" - that might need to be documented here and its overloads, too.

Reply:
I think this one is clear its the class progid, not the interface progid